src/Abstractions/DurableTaskRegistry.Entities.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Container for registered <see cref="ITaskOrchestrator" />, <see cref="ITaskActivity" />,
+/// and <see cref="ITaskEntity"/> implementations.
+/// </summary>
+public partial class DurableTaskRegistry
```

Comment:
nitpick: in comments, we can group functions by registration types to enhance readability? ex.
// --- Type(Factory)-based Registration ---
// --- Generic Type-based Registration ---
// --- Singleton Instance Registration ---

Reply:
I don't think this is necessary. There are only 6 methods and they're all quite short, so it's not difficult to read as-is.